### PR TITLE
Allow integrations tests to run with selected backends

### PR DESCRIPTION
**Pull request by Smit-create**
After merging this we won't require `integration_tests/run_tests.sh`

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
Please also try to support this commit https://github.com/lcompilers/lpython/pull/1262/commits/06631264054a21518d344a4214dfa97b6f63f403 in this new `integration_tests/run_tests_new.py`.

**Comment by Shaikh-Ubaid** [comment]
>Please also try to support this commit https://github.com/lcompilers/lpython/commit/06631264054a21518d344a4214dfa97b6f63f403 in this new integration_tests/run_tests_new.py.

Or, maybe the `CI` itself can select which backends to run on which platforms with the new `integration_tests/run_tests_new.py`.

**Comment by Smit-create** [comment]
The error is that the runtime_library path for `lpython` has some issues:
```
ltypes_path: /usr/share/miniconda/envs/test/bin/../share/lpython/lib/ltypes/ltypes.py
read_status: 0
ltypes_path: /home/runner/work/lpython/lpython/integration_tests/ltypes/ltypes.py
read_status: 0
```

**Comment by Thirumalai-Shaktivel** [comment]
I'm trying to debug the issue in here. As you said, I think it's something related to path!

**Comment by Thirumalai-Shaktivel** [comment]
So, this works in your local machine, right?

**Comment by Smit-create** [comment]
> So, this works in your local machine, right?

Yes, it does!

**Comment by Smit-create** [comment]
> I'm trying to debug the issue here. As you said, I think it's something related to the path!

That has something to do with `runtime_library_path` in `python_ast_to_asr.cpp`

**Comment by Thirumalai-Shaktivel** [comment]
Yup, the problem is with these two:
```py
ltypes_path: /usr/share/miniconda/envs/test/bin/../share/lpython/lib/ltypes/ltypes.py
read_status: 0
ltypes_path: /home/runner/work/lpython/lpython/integration_tests/ltypes/ltypes.py
read_status: 0
```
It should be:
```py
ltypes_path: /home/thirumalai/Open_Source/lpython/src/bin/../runtime/ltypes/ltypes.py
read_status: 1
ltypes_path: /home/thirumalai/Open_Source/lpython/src/bin/../runtime/ltypes/ltypes.py
read_status: 1
```

**Comment by Thirumalai-Shaktivel** [comment]
> That has something to do with runtime_library_path in python_ast_to_asr.cpp

But, why?
You didn't make any changes in `python_ast_to_asr.cpp`, right?

**Comment by Thirumalai-Shaktivel** [comment]
Here is the issue:
```py
Path: /usr/share/miniconda/envs/test/bin/lpython
DirName: /usr/share/miniconda/envs/test/bin
```
which is the lpython executable path.
Here:
https://github.com/lcompilers/lpython/blob/d4a6d5360ca520d9a25d658e6660cb39a2eda4e2/src/lpython/utils.cpp#L53-L75

File: integration_tests/CMakeLists.txt
Comment (Thirumalai-Shaktivel) [reply integration_tests/CMakeLists.txt:63]: Perfect, All the tests passed!

**Comment by Smit-create** [comment]
Thanks @Thirumalai-Shaktivel

File: integration_tests/CMakeLists.txt
Comment (Smit-create) [reply integration_tests/CMakeLists.txt:63]: Thanks, I just have a question that how does that work in current `main` then?

Comment (Smit-create) [reply integration_tests/CMakeLists.txt:63]: Ah, I get the exact bug now. It has to do with the `PATH` variable. In the current main branch we have the following:
```
% export PATH="$(pwd)/../src/bin:$PATH"
```
It adds the `lpython` executable path first and during the path search, we hit the expected path first.

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Comment by Smit-create** [comment]
@certik @czgdp1807 Please review this PR.

**Review by Shaikh-Ubaid — APPROVED** [review]

**Review by czgdp1807 — APPROVED** [review]
This is fine but I would suggest to wait until #1271 to be merged.

**Pull request merged**
